ext/MitsubaLoader/CElementMatrix.h

```diff
@@ -0,0 +1,44 @@
+﻿#ifndef __C_ELEMENT_MATRIX_H_INCLUDED__
+#define __C_ELEMENT_MATRIX_H_INCLUDED__
+
+#include "IElement.h"
+#include "irrlicht.h"
+
+namespace irr { namespace ext { namespace MitsubaLoader {
+
+class CElementMatrix : public IElement
+{
+public:
+	enum class Type
+	{
+		ARBITRARY,
+		TRANSLATION,
+		ROTATION,
+		SCALE
+	};
+
+public:
+	CElementMatrix(CElementMatrix::Type _type)
+		:type(_type) {};
+
+
+	virtual bool processAttributes(const char** _args) override;
+	virtual bool onEndTag(asset::IAssetManager& _assetManager, IElement* _parent) override;
+	virtual IElement::Type getType() const override { return IElement::Type::MATRIX;  };
+	virtual std::string getLogName() const override { return "matrix"; };
+
+	const core::matrix4SIMD getMatrix() const { return matrix; }
+
+private:
+	core::matrix4SIMD matrix;
+	const CElementMatrix::Type type;
+
+private:
+	std::pair<bool, core::matrix4SIMD> getMatrixFromString(std::string _str) const;
```

Comment:
looks like a candidate for a static function ?

Reply:
I dont think I will need that anywhere else

Reply:
if it does not need the internal state of an object (does not need `this`) if should always be a static.

Bonus point, you can have a private static.